src/mod/applications/mod_callcenter/mod_callcenter.c

```diff
@@ -1678,6 +1761,68 @@ static switch_status_t playback_array(switch_core_session_t *session, const char
 
 	return status;
 }
+static void cc_agent_noanswer_count_status_update(const char *agent_name, const char *member_uuid, const char *member_session_uuid, switch_call_cause_t hangupcause, cc_agent_status_t newstatus, int count, int max)
+{
+	char *column_count = NULL;
+	char *event_cc_action = NULL; 
+	char *event_cc_header_count = NULL; 
+	char *event_cc_header_status = NULL;
+	switch_event_t *event = NULL;
+	char *sql = NULL;
+
+	/* set db column to use and header events */
+	switch (hangupcause) {
+		/* Busy: Do Not Disturb, Circuit congestion */
+		case SWITCH_CAUSE_NORMAL_CIRCUIT_CONGESTION:
+		case SWITCH_CAUSE_USER_BUSY:
+			column_count = "busy_count";
+			event_cc_action = "agent-max-busy";
+			event_cc_header_count = "CC-Agent-Busy-Count";
+			event_cc_header_status = "CC-Agent-Busy-Status";
+			break;
+		/* Reject: User rejected the call */
+		case SWITCH_CAUSE_CALL_REJECTED:
+			column_count = "rejected_count";
+			event_cc_action = "agent-max-rejected";
+			event_cc_header_count = "CC-Agent-Rejected-Count";
+			event_cc_header_status = "CC-Agent-Rejected-Status";
+			break;
+		/* Protection againts super fast loop due to unregistrer */
+		case SWITCH_CAUSE_USER_NOT_REGISTERED:
+			column_count = "not_registered_count";
+			event_cc_action = "agent-max-not-registered";
+			event_cc_header_count = "CC-Agent-Not-Registered-Count";
+			event_cc_header_status = "CC-Agent-Not-Registered-Status";
+			break;
+		/* No answer: Destination does not answer for some other reason */
+		default:
+			column_count = "no_answer_count";
+			event_cc_action = "agent-max-no-answer";
+			event_cc_header_count = "CC-Agent-No-Answer-Count";
+			event_cc_header_status = "CC-Agent-No-Answer-Status";
+			break;
+	}
+
+	/* Update Agent column count */
+	sql = switch_mprintf("UPDATE agents SET %q = %q + 1 WHERE name = '%q';", column_count, column_count, agent_name);
```

Comment:
```suggestion
	sql = switch_mprintf("UPDATE agents SET %s = %s + 1 WHERE name = '%q';", column_count, column_count, agent_name);
```
Very little performance increase ;)  If you rather have %q then it wouldn't bother me